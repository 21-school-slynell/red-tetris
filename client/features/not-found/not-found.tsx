import { Button, Grid, makeStyles, Paper, Typography } from '@material-ui/core';
import React, { FC, memo } from 'react';
import { isServer } from 'client/core/store';

const useStyles = makeStyles({
  root: {
    maxHeight: 1000,
    minHeight: 600,
    display: 'flex',
    position: 'relative',
    flexDirection: 'column',
    justifyContent: 'center',
    padding: 24,
  },
  grid: {
    display: 'grid',
    gap: 16,
  },
});

export const WrapperNotFound: FC = memo(() => {
  const classes = useStyles();

  return (
    <Paper className={classes.root}>
      <Grid xs={12} md={12} className={classes.grid}>
        <Typography variant="h5" align="center">
          Unfortunately, the game has not been found for you
        </Typography>
        <Button
          href="/"
          color="primary"
          style={{ textDecoration: 'underline' }}
        >
          go to main
        </Button>
      </Grid>
    </Paper>
  );
});

const NullFC: FC = () => <></>;

// export const Game = withAuth(!isServer ? WrapperGame : NullFC);
export const NotFound = !isServer ? WrapperNotFound : NullFC;
