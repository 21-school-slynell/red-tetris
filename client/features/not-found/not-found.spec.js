/* eslint-disable react/jsx-filename-extension */
/* eslint-disable no-undef */
/* eslint-disable import/no-unresolved */
import React from 'react';
import renderer from 'react-test-renderer';
import { WrapperNotFound } from './not-found';

describe('WrapperNotFound', () => {
  test('Снапшот для страницы not found', async () => {
    const tree = renderer.create(<WrapperNotFound open />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
