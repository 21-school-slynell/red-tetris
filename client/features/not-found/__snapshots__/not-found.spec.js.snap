// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WrapperNotFound Снапшот для страницы not found 1`] = `
<div
  className="MuiPaper-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiGrid-root makeStyles-grid-2 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter"
    >
      Unfortunately, the game has not been found for you
    </h5>
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
      href="/"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "textDecoration": "underline",
        }
      }
      tabIndex={0}
    >
      <span
        className="MuiButton-label"
      >
        go to main
      </span>
    </a>
  </div>
</div>
`;
